src/test/java/org/gridsuite/shortcircuit/server/repositories/FaultResultRepositoryTest.java

```diff
@@ -169,7 +210,109 @@ private Stream<Arguments> provideOrEqualsNestedFieldsFilters() {
+    }
+
+    private Stream<Arguments> provideFeederFieldsContainFiltersAndSorted() {
+        return Stream.of(
+            Arguments.of(
+                resultMagnitudeEntity,
+                List.of(
+                    new ResourceFilter(ResourceFilter.DataType.TEXT, ResourceFilter.Type.CONTAINS, "1", "feederResults.connectableId")),
+                    Sort.by(
+                            new Sort.Order(Sort.Direction.ASC, "current"),
+                            new Sort.Order(Sort.Direction.DESC, "feederResults.connectableId")),
+                List.of(List.of(FEEDER_RESULT_1), List.of(FEEDER_RESULT_4, FEEDER_RESULT_1))),
+            Arguments.of(
+                resultMagnitudeEntity,
+                List.of(
+                    new ResourceFilter(ResourceFilter.DataType.TEXT, ResourceFilter.Type.CONTAINS, "NN_I", "feederResults.connectableId"),
+                    new ResourceFilter(ResourceFilter.DataType.TEXT, ResourceFilter.Type.CONTAINS, "1", "feederResults.connectableId")),
+                    Sort.by(
+                            new Sort.Order(Sort.Direction.ASC, "current"),
+                            new Sort.Order(Sort.Direction.DESC, "feederResults.connectableId")),
+                List.of(
+                    List.of(FEEDER_RESULT_1), List.of(FEEDER_RESULT_1)
+                )),
+            Arguments.of(
+                resultMagnitudeEntity,
+                List.of(
+                        new ResourceFilter(ResourceFilter.DataType.TEXT, ResourceFilter.Type.CONTAINS, "OUPS", "feederResults.connectableId")),
+                Sort.by(
+                        new Sort.Order(Sort.Direction.ASC, "current"),
+                        new Sort.Order(Sort.Direction.DESC, "feederResults.connectableId")),
+                List.of()
+            )
+        );
+    }
+
+    private Stream<Arguments> provideFeederFieldsUnfilteredSortedAscAndDesc() {
+        return Stream.of(
+            Arguments.of(
+                resultMagnitudeEntity,
+                List.of(),
+                    Sort.by(
+                            new Sort.Order(Sort.Direction.ASC, "current"),
+                            new Sort.Order(Sort.Direction.DESC, "feederResults.connectableId")),
+                List.of(
+                    List.of(FEEDER_RESULT_3, FEEDER_RESULT_1),
+                    List.of(FEEDER_RESULT_4, FEEDER_RESULT_3, FEEDER_RESULT_2, FEEDER_RESULT_1),
+                    List.of()
+                )),
+            Arguments.of(
+                resultMagnitudeEntity,
+                List.of(),
+                    Sort.by(
+                            new Sort.Order(Sort.Direction.ASC, "current"),
+                            new Sort.Order(Sort.Direction.ASC, "feederResults.connectableId")),
+                List.of(
+                    List.of(FEEDER_RESULT_1, FEEDER_RESULT_3),
+                    List.of(FEEDER_RESULT_1, FEEDER_RESULT_2, FEEDER_RESULT_3, FEEDER_RESULT_4),
+                    List.of()
```

Comment:
Maybe there is something I did not get here, but if list of children is empty, parent should be removed
Why is there an empty list in expected result ? Same for some other tests